### PR TITLE
preserve quote label

### PR DESCRIPTION
**Pull request by smtakeda**
Fixing #57.

### CODE REVIEW

**Comment by smtakeda** [comment]
@villebro, would you please review this change?

**Comment by villebro** [comment]
@smtakeda sure, would be happy to do so, will review today.

File: test/test_compiler.py [hunk 1]
```diff
@@ -46,11 +45,49 @@ def test_multi_table_delete_multiple(self):
...
+FROM "abc" GROUP BY colname"""
+
+    # Not quote label
+    col = column('colname').label('alias')
+    sel_from_tbl = select([col]).group_by(col).select_from(
+        table(quoted_name('abc', True)))
+    compiled_result = sel_from_tbl.compile(engine_testaccount)
+
+    assert str(compiled_result) == """SELECT colname AS alias 
+FROM "abc" GROUP BY colname"""
```
Comment (villebro) [inline test/test_compiler.py]: For the sake of clarity, I would not quote the table name in this case:

```python
    # Not quote label
    col = column('colname').label('alias')
    sel_from_tbl = select([col]).group_by(col).select_from('abc')
    compiled_result = sel_from_tbl.compile(engine_testaccount)
     assert str(compiled_result) == """SELECT colname AS alias 
FROM abc GROUP BY colname"""
```

File: test/test_compiler.py [hunk 2]
```diff
@@ -46,11 +45,49 @@ def test_multi_table_delete_multiple(self):
...
+
+    # Not quote label in capital case
+    col = column('colname').label('ALIAS')
+    sel_from_tbl = select([col]).group_by(col).select_from(
+        table(quoted_name('abc', True)))
+    compiled_result = sel_from_tbl.compile(engine_testaccount)
+
+    assert str(compiled_result) == """SELECT colname AS "ALIAS" 
+FROM "abc" GROUP BY colname"""
+
```
Comment (villebro) [inline test/test_compiler.py]: I believe having a mixed-case test case might be better, as all uppercase in this case would in fact not impact the results (do correct me if I am wrong). So perhaps the following might be more important to test. Also here remove quoting from table name:

```python
    # Not quote label in mixed case
    col = column('colname').label('Alias')
    sel_from_tbl = select([col]).group_by(col).select_from('abc')
    compiled_result = sel_from_tbl.compile(engine_testaccount)
     assert str(compiled_result) == """SELECT colname AS "Alias" 
FROM abc GROUP BY colname"""
```

**Comment by villebro** [comment]
@smtakeda I added a few comments, but overall LGTM.

File: test/test_compiler.py [hunk 3]
```diff
@@ -46,11 +45,49 @@ def test_multi_table_delete_multiple(self):
...
+def test_quoted_name_label(engine_testaccount):
+    # quote label
+    col = column('colname').label(quoted_name('alias', True))
+    sel_from_tbl = select([col]).group_by(col).select_from(
+        table(quoted_name('abc', True)))
+    compiled_result = sel_from_tbl.compile(engine_testaccount)
+
+    assert str(compiled_result) == """SELECT colname AS "alias" 
+FROM "abc" GROUP BY colname"""
+
```
Comment (villebro) [inline test/test_compiler.py]: For the sake of clarity, I would not quote the table name in this case:

```python
    # quote label
    col = column('colname').label(quoted_name('alias', True))
    sel_from_tbl = select([col]).group_by(col).select_from(
        table('abc'))
    compiled_result = sel_from_tbl.compile(engine_testaccount)
     assert str(compiled_result) == """SELECT colname AS "alias" 
FROM abc GROUP BY colname"""
```

File: test/test_compiler.py [hunk 4]
```diff
@@ -46,11 +45,49 @@ def test_multi_table_delete_multiple(self):
...
+    # Not quote table
+    col = column('colname').label('alias')
+    sel_from_tbl = select([col]).group_by(col).select_from(
+        table('abc'))
+    compiled_result = sel_from_tbl.compile(engine_testaccount)
+
+    assert str(compiled_result) == """SELECT colname AS alias 
+FROM abc GROUP BY colname"""
```
Comment (villebro) [inline test/test_compiler.py]: This test case I would remove, unless there is an explicit need to test quoting of table names.

File: test/test_compiler.py (hunk 3 above)
Comment (smtakeda) [reply test/test_compiler.py]: ok, but your test case in #57 did the quote the table name?

File: test/test_compiler.py (hunk 2 above)
Comment (smtakeda) [reply test/test_compiler.py]: makes sense.

File: test/test_compiler.py (hunk 3 above)
Comment (villebro) [reply test/test_compiler.py]: Yeah, I don't remember exactly what my reasoning was, but the test case was written in just a few minutes and I think my point was to show that it was quoting the table name, but not the label. But to keep things as modular as possible, I would keep the label quoting test cases separate from any potential table name quoting tests.

File: test/test_compiler.py (hunk 4 above)
Comment (smtakeda) [reply test/test_compiler.py]: ok, sounds good.

File: test/test_compiler.py (hunk 3 above)
Comment (smtakeda) [reply test/test_compiler.py]: Fair enough.

**Comment by smtakeda** [comment]
@villebro, thanks for reviewing the code!

**Comment by villebro** [comment]
Thanks @smtakeda, I like the latest changes!

**Comment by tomsej** [comment]
Hi, thanks for this! Do you know when v1.1.4 will be released?

**Comment by smtakeda** [comment]
v1.1.4 will be released next Tuesday. Thanks.

**Pull request merged**